src/fxos/readGeckoCommit.js

```diff
@@ -0,0 +1,75 @@
+let assert = require('assert');
+let debug = console.log.bind(console, '[fxos/readGeckoCommit]');
+let denodeify = require('denodeify');
+let findInStream = require('../find-in-stream');
+let fs = require('mz/fs');
+let path = require('path');
+let sax = require('sax');
+let tmpdir = denodeify(require('tmp').dir);
+
+const SOURCES_XML = '/system/sources.xml';
+const APPLICATION_INI = '/system/b2g/application.ini';
+const PLATFORM_INI = '/system/b2g/platform.ini';
+
+async function pullIniFile(adb, local, file) {
+  await adb.pull(file, local);
+  return path.join(local, path.basename(file));
+}
+
+function matchSourceStamp(item) {
+  return item
+    .split('\n')
+    .find(line => line.includes('SourceStamp'));
+}
+
+async function readGeckoRevisionFromIni(iniFile) {
+  let stream = fs.createReadStream(iniFile, {encoding: 'utf8'});
+  let line = await findInStream(stream, 'data', matchSourceStamp);
```

Comment:
If we did that we would need to reorder the args. If we did that I'd want predicate first.